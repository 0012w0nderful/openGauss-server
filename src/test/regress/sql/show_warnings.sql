drop database if exists db_show_warnings;
create database db_show_warnings dbcompatibility 'b';
\c db_show_warnings

show max_error_count;
show sql_note;
create table test(id int, name varchar default 11);
insert into t1 values(1,'test');
show warnings limit 1;
show errors limit 1;
show count(*) warnings;
show count(*) errors;
set max_error_count = 0;
show errors;
set max_error_count = 64;

CREATE OR REPLACE FUNCTION TEST_FUNC(tempdata char) RETURNS VOID AS $$
BEGIN
	raise info'TEST CHAR VALUE IS %',tempdata;  
END;
$$ LANGUAGE plpgsql;
select TEST_FUNC('abc'::clob);

show warnings;
set sql_note=false;
select TEST_FUNC('abc'::clob);
show warnings;

\c postgres
drop database if exists db_show_warnings;
