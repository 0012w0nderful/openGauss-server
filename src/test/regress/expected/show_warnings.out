drop database if exists db_show_warnings;
NOTICE:  database "db_show_warnings" does not exist, skipping
create database db_show_warnings dbcompatibility 'b';
\c db_show_warnings
show max_error_count;
 max_error_count 
-----------------
 64
(1 row)

show sql_note;
 sql_note 
----------
 on
(1 row)

create table test(id int, name varchar default 11);
insert into t1 values(1,'test');
ERROR:  relation "t1" does not exist on datanode1
LINE 1: insert into t1 values(1,'test');
                    ^
show warnings limit 1;
 level |   code   |                  message                  
-------+----------+-------------------------------------------
 Error | 16908420 | relation "t1" does not exist on datanode1
(1 row)

show errors limit 1;
 level |   code   |                  message                  
-------+----------+-------------------------------------------
 Error | 16908420 | relation "t1" does not exist on datanode1
(1 row)

show count(*) warnings;
 count 
-------
     1
(1 row)

show count(*) errors;
 count 
-------
     1
(1 row)

set max_error_count = 0;
show errors;
 level | code | message 
-------+------+---------
(0 rows)

set max_error_count = 64;
CREATE OR REPLACE FUNCTION TEST_FUNC(tempdata char) RETURNS VOID AS $$
BEGIN
	raise info'TEST CHAR VALUE IS %',tempdata;  
END;
$$ LANGUAGE plpgsql;
select TEST_FUNC('abc'::clob);
INFO:  TEST CHAR VALUE IS abc
CONTEXT:  referenced column: test_func
 test_func 
-----------
 
(1 row)

show warnings;
 level | code |        message         
-------+------+------------------------
 Note  |    0 | TEST CHAR VALUE IS abc
(1 row)

set sql_note=false;
select TEST_FUNC('abc'::clob);
INFO:  TEST CHAR VALUE IS abc
CONTEXT:  referenced column: test_func
 test_func 
-----------
 
(1 row)

show warnings;
 level | code | message 
-------+------+---------
(0 rows)

\c postgres
drop database if exists db_show_warnings;
